CONTRIBUTING.md

```diff
@@ -1,120 +1,116 @@
 
-**3.** Improve Documentation
 
+## Setup
 
-## Setup 🚀
-You need to set up Textualize to make your contribution. Textual requires Python 3.7 or later (if you have a choice, pick the most recent Python). Textual runs on Linux, macOS, Windows, and probably any OS where Python also runs.
+To make a code or documentation contribution you will need to set up Textual locally.
+You can follow these steps:
 
-### Installation
+ 1. Make sure you have Poetry installed ([see instructions here](https://python-poetry.org))
+ 2. Clone the Textual repository
+ 3. Run `poetry shell` to create a virtual environment for the dependencies
+ 4. Run `poetry install` to install all dependencies
+ 5. Make sure the latest version of Textual was installed by running the command `textual --version`
+ 6. Install the pre-commit hooks with the command `pre-commit install`
 
-**Install Texualize via pip:**
-```bash
-pip install textual
-```
-**Install [Poetry](https://python-poetry.org/)**
-```bash
-curl -sSL https://install.python-poetry.org | python3 -
-```
-**To install all dependencies, run:**
-```bash
-poetry install --all 
-```
-**Make sure everything works fine:**
-```bash
-textual --version
-```
-### Demo
+## Demo
 
-Once you have Textual installed, run the following to get an impression of what it can do:
+Once you have Textual installed, run the Textual demo to get an impression of what Textual can do and to double check that everything was installed correctly:
 
 ```bash
 python -m textual
 ```
-If Texualize is installed, you should see this:
-<img width="848" alt="demo" src="https://github.com/clairecharles/textual/assets/67120042/62fd53c0-7ad6-4429-8751-5a713180b836">
 
-## Make contribution
-**1.** Fork [this](repo) repository.
+## Guidelines
 
-**2.** Clone the forked repository.
+- Make sure to read the issue instructions carefully. If something isn't clear, ask for clarification!
 
-```bash
-git clone https://github.com/<your-username>/textual.git
-```
+- Add docstrings to all of your code (functions, methods, classes, ...). The codebase should have enough examples for you to copy from.
+
+- Write tests for your code.
+  - If you are fixing a bug, make sure to add regression tests that link to the original issue.
+  - If you are implementing a visual element, make sure to add _snapshot tests_. [See below](#snapshot-testing) for more details.
 
-**3.** Navigate to the project directory.
+## Before opening a PR
 
-```bash
-cd textual
-```
+Before you open your PR, please go through this checklist and make sure you've checked all the items that apply:
 
-**4.** Create a new [pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+ - [ ] Update the `CHANGELOG.md`
+ - [ ] Format your code with black (`make format`)
+ - [ ] All your code has docstrings in the style of the rest of the codebase
+ - [ ] Your code passes all tests (`make test`)
+ - [ ] You added documentation under `docs/`
```

Comment:
I don't think we should encourge writing to the docs directory. Not all developers are great at writing docs. Docstrings should cover most API changes.